clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -788,36 +788,27 @@ protected synchronized void markCoordinatorUnknown(boolean isDisconnected) {
     }
 
     /**
-     * Get the current generation state if the group is stable.
-     * @return the current generation or null if the group is unjoined/rebalancing
+     * Get the current generation state
+     *
+     * @return the current generation
      */
     protected synchronized Generation generation() {
-        if (this.state != MemberState.STABLE)
-            return null;
         return generation;
     }
 
-    protected synchronized String memberId() {
-        return generation == null ? JoinGroupRequest.UNKNOWN_MEMBER_ID :
```

Comment:
While working on the PR, I realize that now generation would never be null, so I updated the logic and also the unit tests accordingly.

Reply:
Nice cleanup, I notice btw in `WorkerCoordinator#memberId` we call `Generation generation = generation();` followed by a check for null. I'm guessing this should be updated to use `generationIfStable`?

Reply:
Yes I'm already using `generationIfStable`

Reply:
Ah sorry, didn't see it there at the bottom 😅